### ISSUE TITLE
Modify_slots does not work with subclasses

### ISSUE THREAD

**Issue by lstarin** [noxdafox/clipspy#47]
Hi, I have a problem when I use modify_slots on an instance of a subclass. Hope you can help me.
Using your code as a basis:

```python
import clips
env = clips.Environment()

defclass = """
(defclass MyClass (is-a USER)
  (slot One)
  (slot Two))
"""

env.build(defclass)

defclass = env.find_class("MyClass")
instance = defclass.make_instance("MyInstance", One=1, Two=2)
fact = env.assert_string('(foo bar baz)')
instance.modify_slots(One=fact)

defclass = """
(defclass SUBMyClass (is-a MyClass)
  (slot Three))
"""

env.build(defclass)

defclass = env.find_class("SUBMyClass")
instance = defclass.make_instance("SUBMyInstance", One=1)
for i in env.instances():
    print(i)
```

This shows: 
```
[MyInstance] of MyClass (One <Fact-1>) (Two 2)
[SUBMyInstance] of SUBMyClass (One 1) (Two nil) (Three nil)
```

Then, when I do this: 

```python
instance.modify_slots(Three=3)
```

The kernel dies...
Any idea of what I'm doing wrong?
Thanks in advance! And for your work!
Laura

**Comment by noxdafox** [comment]
I can confirm the issue. Nevertheless, the problem lies within the CLIPS engine itself and not CLIPSPy.

Hence, it will take some time to get it fixed as it's out of the scope of this project.

In the meantime, you can workaround the issue by using the `modify-instance` builtin function.
```python
instance = defclass.make_instance("SUBMyInstance", One=1)
env.eval(f'(modify-instance [{instance.name}] (Three 3))')

for i in env.instances():
    print(i)
```

**Comment by noxdafox** [comment]
Issue reported to upstream project:
https://sourceforge.net/p/clipsrules/tickets/69/

**Comment by noxdafox** [comment]
Issue fixed in release `1.0.1`.
